src/entities/entity.cpp

```diff
@@ -0,0 +1,114 @@
+/***
+ * Maracatronics Robotics
+ * Federal University of Pernambuco (UFPE) at Recife
+ * http://www.maracatronics.com/
+ *
+ * This file is part of Armorial project.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ***/
+
+#include "entity.h"
+
+Entity::Entity(EntityType type) {
+    _entityType = type;
+    _entityPriority = 0; // default priority is 0
+    _loopFrequency = 60; // default loop frequency is 60
+    _isEnabled = true;   // enabling by default
+}
+
+void Entity::run(){
+    initialization();
+
+    while(isEnabled()) {
+        startTimer();
+        if(isEnabled()) {
+            loop();
+        }
+        stopTimer();
+
+        long rest = getRemainingTime();
+        if(rest >= 0) {
+            msleep(rest);
+        }
+    }
+
+    finalization();
+}
+
+void Entity::setLoopFrequency(int hz) {
+    _mutexLoopTime.lock();
+    _loopFrequency = hz;
+    _mutexLoopTime.unlock();
+}
+
+void Entity::setPriority(int priority) {
+    _mutexPriority.lock();
+    _entityPriority = priority;
+    _mutexPriority.unlock();
+}
+
+void Entity::enableEntity() {
+    _mutexEnabled.lock();
+    _isEnabled = true;
+    _mutexEnabled.unlock();
+}
+
+void Entity::stopEntity() {
```

Comment:
Use this method to threads that don't need to run (like our offline players or enemy players)